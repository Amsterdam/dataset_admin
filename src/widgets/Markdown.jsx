import React from 'react';
import PropTypes from 'prop-types';

const Markdown = props => (
  <div>
    <textarea
      id={props.id}
      className="markdown-field"
      placeholder={props.placeholder}
      disabled={props.disabled}
      readOnly={props.readonly}
      label={props.label}
      value={props.value}
    />
    <div>{console.log(props, props.options, props.schema)}</div>
  </div>
);

Markdown.defaultProps = {
  label: '',
  placeholder: '',
  options: {},
  schema: {},
  value: ''
};

Markdown.propTypes = {
  // autofocus: PropTypes.bool.isRequired,
  disabled: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  options: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  readonly: PropTypes.bool.isRequired,
  // required: PropTypes.bool.isRequired,
  schema: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  value: PropTypes.string
};

export default Markdown;
