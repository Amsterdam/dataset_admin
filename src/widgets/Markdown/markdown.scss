@import "../../styling/config.scss";

.markdown {
  .form-control {
    height: 148px !important;
  }

  &__preview {
    background-color: $white-four;
    height: auto;
    margin-top: 16px;
    overflow: auto;
    padding: 4px 10px;
  }
}
