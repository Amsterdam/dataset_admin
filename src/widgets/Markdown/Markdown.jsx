import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ReactMarkdown from 'react-markdown';

import './markdown.scss';

class Markdown extends Component {
  constructor(props) {
    super(props);

    this.state = {
      value: props.value
    };

    this.handleChange = this.handleChange.bind(this);
  }

  componentWillReceiveProps(props) {
    this.setState({
      value: props.value
    });
  }

  handleChange(event) {
    this.setState({ value: event.target.value });
    this.props.onChange(event.target.value || undefined);
  }

  render() {
    const { value } = this.state;

    return (
      <div className="markdown">
        <textarea
          id={this.props.id}
          className="form-control"
          placeholder={this.props.placeholder}
          disabled={this.props.disabled}
          readOnly={this.props.readonly}
          rows={this.props.options.rows}
          label={this.props.label}
          required={this.props.required}
          value={value}

          onChange={this.handleChange}
        />
        <ReactMarkdown
          source={value || 'preview'}
          className="markdown__preview"
        />
      </div>
    );
  }
}

Markdown.defaultProps = {
  disabled: false,
  label: '',
  placeholder: '',
  options: {},
  readonly: false,
  required: false,
  rows: 3,
  value: '',
  onChange: () => {}
};

Markdown.propTypes = {
  disabled: PropTypes.bool,
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  options: PropTypes.object,
  readonly: PropTypes.bool,
  rows: PropTypes.number,
  required: PropTypes.bool,
  value: PropTypes.string,
  onChange: PropTypes.func
};

export default Markdown;
