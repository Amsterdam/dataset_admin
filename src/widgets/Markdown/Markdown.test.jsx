import React from 'react';

import Markdown from './Markdown';

describe('The Markdown component (react-jsonschema-forms custom widget)', () => {
  let wrap;

  beforeEach(() => {
    wrap = shallow(<Markdown id="very-unique-id" />);
  });

  describe('renders component', () => {
    it('renders default', () => {
      expect(wrap).toMatchSnapshot();
    });

    it('renders new value', () => {
      wrap.setProps({
        value: 'foo'
      });

      expect(wrap).toMatchSnapshot();
    });
  });

  describe('triggering events', () => {
    it('updates the value after change', () => {
      const event = { target: { value: 'bar' } };

      wrap.props().children[0].props.onChange(event);
      expect(wrap).toMatchSnapshot();
    });

    it('updates the value to undefined when string is empty', () => {
      const event = { target: { value: '' } };

      wrap.props().children[0].props.onChange(event);
      expect(wrap).toMatchSnapshot();
    });
  });
});
