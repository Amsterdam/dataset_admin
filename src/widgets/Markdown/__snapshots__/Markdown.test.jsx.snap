// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Markdown component (react-jsonschema-forms custom widget) renders component renders default 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Markdown
    disabled={false}
    id="very-unique-id"
    label=""
    onChange={[Function]}
    options={Object {}}
    placeholder=""
    readonly={false}
    required={false}
    rows={3}
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <textarea
          className="form-control"
          disabled={false}
          id="very-unique-id"
          label=""
          onChange={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          rows={undefined}
          value=""
        />,
        <ReactMarkdown
          className="markdown__preview"
          escapeHtml={true}
          renderers={Object {}}
          skipHtml={false}
          source="preview"
          transformLinkUri={[Function]}
        />,
      ],
      "className": "markdown",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control",
          "disabled": false,
          "id": "very-unique-id",
          "label": "",
          "onChange": [Function],
          "placeholder": "",
          "readOnly": false,
          "required": false,
          "rows": undefined,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "markdown__preview",
          "escapeHtml": true,
          "renderers": Object {},
          "skipHtml": false,
          "source": "preview",
          "transformLinkUri": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <textarea
            className="form-control"
            disabled={false}
            id="very-unique-id"
            label=""
            onChange={[Function]}
            placeholder=""
            readOnly={false}
            required={false}
            rows={undefined}
            value=""
          />,
          <ReactMarkdown
            className="markdown__preview"
            escapeHtml={true}
            renderers={Object {}}
            skipHtml={false}
            source="preview"
            transformLinkUri={[Function]}
          />,
        ],
        "className": "markdown",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "disabled": false,
            "id": "very-unique-id",
            "label": "",
            "onChange": [Function],
            "placeholder": "",
            "readOnly": false,
            "required": false,
            "rows": undefined,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "markdown__preview",
            "escapeHtml": true,
            "renderers": Object {},
            "skipHtml": false,
            "source": "preview",
            "transformLinkUri": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`The Markdown component (react-jsonschema-forms custom widget) renders component renders new value 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Markdown
    disabled={false}
    id="very-unique-id"
    label=""
    onChange={[Function]}
    options={Object {}}
    placeholder=""
    readonly={false}
    required={false}
    rows={3}
    value="foo"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <textarea
          className="form-control"
          disabled={false}
          id="very-unique-id"
          label=""
          onChange={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          rows={undefined}
          value="foo"
        />,
        <ReactMarkdown
          className="markdown__preview"
          escapeHtml={true}
          renderers={Object {}}
          skipHtml={false}
          source="foo"
          transformLinkUri={[Function]}
        />,
      ],
      "className": "markdown",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control",
          "disabled": false,
          "id": "very-unique-id",
          "label": "",
          "onChange": [Function],
          "placeholder": "",
          "readOnly": false,
          "required": false,
          "rows": undefined,
          "value": "foo",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "markdown__preview",
          "escapeHtml": true,
          "renderers": Object {},
          "skipHtml": false,
          "source": "foo",
          "transformLinkUri": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <textarea
            className="form-control"
            disabled={false}
            id="very-unique-id"
            label=""
            onChange={[Function]}
            placeholder=""
            readOnly={false}
            required={false}
            rows={undefined}
            value="foo"
          />,
          <ReactMarkdown
            className="markdown__preview"
            escapeHtml={true}
            renderers={Object {}}
            skipHtml={false}
            source="foo"
            transformLinkUri={[Function]}
          />,
        ],
        "className": "markdown",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "disabled": false,
            "id": "very-unique-id",
            "label": "",
            "onChange": [Function],
            "placeholder": "",
            "readOnly": false,
            "required": false,
            "rows": undefined,
            "value": "foo",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "markdown__preview",
            "escapeHtml": true,
            "renderers": Object {},
            "skipHtml": false,
            "source": "foo",
            "transformLinkUri": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`The Markdown component (react-jsonschema-forms custom widget) triggering events updates the value after change 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Markdown
    disabled={false}
    id="very-unique-id"
    label=""
    onChange={[Function]}
    options={Object {}}
    placeholder=""
    readonly={false}
    required={false}
    rows={3}
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <textarea
          className="form-control"
          disabled={false}
          id="very-unique-id"
          label=""
          onChange={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          rows={undefined}
          value=""
        />,
        <ReactMarkdown
          className="markdown__preview"
          escapeHtml={true}
          renderers={Object {}}
          skipHtml={false}
          source="preview"
          transformLinkUri={[Function]}
        />,
      ],
      "className": "markdown",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control",
          "disabled": false,
          "id": "very-unique-id",
          "label": "",
          "onChange": [Function],
          "placeholder": "",
          "readOnly": false,
          "required": false,
          "rows": undefined,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "markdown__preview",
          "escapeHtml": true,
          "renderers": Object {},
          "skipHtml": false,
          "source": "preview",
          "transformLinkUri": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <textarea
            className="form-control"
            disabled={false}
            id="very-unique-id"
            label=""
            onChange={[Function]}
            placeholder=""
            readOnly={false}
            required={false}
            rows={undefined}
            value=""
          />,
          <ReactMarkdown
            className="markdown__preview"
            escapeHtml={true}
            renderers={Object {}}
            skipHtml={false}
            source="preview"
            transformLinkUri={[Function]}
          />,
        ],
        "className": "markdown",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "disabled": false,
            "id": "very-unique-id",
            "label": "",
            "onChange": [Function],
            "placeholder": "",
            "readOnly": false,
            "required": false,
            "rows": undefined,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "markdown__preview",
            "escapeHtml": true,
            "renderers": Object {},
            "skipHtml": false,
            "source": "preview",
            "transformLinkUri": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`The Markdown component (react-jsonschema-forms custom widget) triggering events updates the value to undefined when string is empty 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Markdown
    disabled={false}
    id="very-unique-id"
    label=""
    onChange={[Function]}
    options={Object {}}
    placeholder=""
    readonly={false}
    required={false}
    rows={3}
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <textarea
          className="form-control"
          disabled={false}
          id="very-unique-id"
          label=""
          onChange={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          rows={undefined}
          value=""
        />,
        <ReactMarkdown
          className="markdown__preview"
          escapeHtml={true}
          renderers={Object {}}
          skipHtml={false}
          source="preview"
          transformLinkUri={[Function]}
        />,
      ],
      "className": "markdown",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control",
          "disabled": false,
          "id": "very-unique-id",
          "label": "",
          "onChange": [Function],
          "placeholder": "",
          "readOnly": false,
          "required": false,
          "rows": undefined,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "markdown__preview",
          "escapeHtml": true,
          "renderers": Object {},
          "skipHtml": false,
          "source": "preview",
          "transformLinkUri": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <textarea
            className="form-control"
            disabled={false}
            id="very-unique-id"
            label=""
            onChange={[Function]}
            placeholder=""
            readOnly={false}
            required={false}
            rows={undefined}
            value=""
          />,
          <ReactMarkdown
            className="markdown__preview"
            escapeHtml={true}
            renderers={Object {}}
            skipHtml={false}
            source="preview"
            transformLinkUri={[Function]}
          />,
        ],
        "className": "markdown",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "disabled": false,
            "id": "very-unique-id",
            "label": "",
            "onChange": [Function],
            "placeholder": "",
            "readOnly": false,
            "required": false,
            "rows": undefined,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "markdown__preview",
            "escapeHtml": true,
            "renderers": Object {},
            "skipHtml": false,
            "source": "preview",
            "transformLinkUri": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
